resource/nodestate_string.go

```diff
@@ -13,11 +13,12 @@ func _() {
 	_ = x[NodeStateConfiguring-2]
 	_ = x[NodeStateReady-3]
 	_ = x[NodeStateRemoving-4]
+	_ = x[NodeStateUnhealthy-5]
 }
 
-const _NodeState_name = "UnknownUnconfiguredConfiguringReadyRemoving"
+const _NodeState_name = "UnknownUnconfiguredConfiguringReadyRemovingUnhealthy"
```

Comment:
What is this? Is this code generation? Does our `Makefile` regenerate this? Does a human have to run a command? What's the input file? Where's a link to the tools documentation?

Reply:
This file is indeed auto-generated using [the stringer tool](https://pkg.go.dev/golang.org/x/tools/cmd/stringer). the `Makefile` command that does this is `make generate-go` - currently a human has to run this command whenever the underlying enum changes - CI will fail otherwise, similarly to how CI fails if `make lint-go` produces any changes.

The input file is the source code - [this directive](https://github.com/viamrobotics/rdk/blob/39ab463dd068b45beec73d44cab04c3a473c45c0/resource/graph_node.go#L14) tells the generator what Enum needs to be stringified and how.

The stringer tool [is imported here](https://github.com/viamrobotics/rdk/blob/39ab463dd068b45beec73d44cab04c3a473c45c0/tools/tools.go#L16)

Will add some documentation in these 3 spots (i.e. Makefile, source, tools import)

Reply:
the most important "doc" change is probably the CI error, since that informs devs that then need to run `make generate-go`: https://github.com/viamrobotics/rdk/pull/4257/commits/dc565ad029a3d7b66ac0692626a092d7d9544a3e

Reply:
Thanks for this! That addition looks great to me. One more small request, can we add a comment [near the NodeState type](https://github.com/viamrobotics/rdk/blob/39ab463dd068b45beec73d44cab04c3a473c45c0/resource/graph_node.go#L17) that mentions that the `go:generate stringer` creates `String` methods for `NodeState` objects?

Reply:
Added near the NodeState typedef as well as in tools.go: https://github.com/viamrobotics/rdk/pull/4257/commits/438b33342d51f25db97120e20e20389ccdfbb23e